### DISCUSSION TITLE
GAP over Lap

### DISCUSSION THREAD

**Discussion by nicZox** [theOehrly/Fast-F1#390] [Q&A]
Hi,
is there anyone who can help me calculate the GAP during the lap? (Like F1-Tempo does)

Thanks

![image](https://github.com/theOehrly/Fast-F1/assets/76594617/b71f0f12-f875-4dbd-aee9-86472d2c903b)

**Comment by edmundhong** [comment]
![image](https://github.com/theOehrly/Fast-F1/assets/76190690/aaaac17b-ce79-4add-bd95-3e8105253c5f)
(df = Session.laps)

Here is the code snippet that I used to estimate the time delta, by cutting the laps into mini sectors and then calculating the average time taken. Lastly, compare the time difference from the best driver. You can explore more at my [repo](https://github.com/edmundhong/formula1-dataanalysis/blob/main/Best%20Lap%20Analysis.ipynb).

**Reply by Casper-Guo** [reply]
You gotta add some of your plots to the example gallery. The track dominance plot is cool and there's a need for examples showcasing the weather data.

**Reply by edmundhong** [reply]
cool, thanks!

**Comment by harningle** [comment]
This is amazing! I've made something similar before, but plots the gaps on the track map (similar to those [head to head video](https://www.youtube.com/watch?v=rcERq7GoMVw)). My mini sector time gaps were calculated in a more sloppy way, but do not seem too crazy compare to the video on F1 youtube channel.

*Mine*
![Screenshot_20230725-153252](https://github.com/theOehrly/Fast-F1/assets/31792725/a9a39189-4151-440f-bede-15e28d488d11)

*F1 youtube*
![Untitled](https://github.com/theOehrly/Fast-F1/assets/31792725/2bdf10ea-cced-45fd-9198-9b7f0eb3c3a2)
